codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -384,4 +373,71 @@ protected ZlibWrapper determineWrapper(String acceptEncoding) {
         }
         return null;
     }
+
+    private CompressionEncoderFactory getEncoderFactory(String contentEncoding) {
+        for (CompressionEncoderFactory encoderFactory: factories) {
+            if (encoderFactory.name().equals(contentEncoding)) {
+                return encoderFactory;
+            }
+        }
+
+        return null;
+    }
+
+    private class GzipEncoderFactory implements CompressionEncoderFactory {
```

Comment:
final static

Reply:
This can be `final` but not `static`, because `CompressionOptions` is referenced here, which is an instance variable

Reply:
How about just pass it in in the constructor ?

Reply:
I tried this way, but when instantiating `MessageToByteEncoder`, what is needed is not `CompressionOptions`, but its subclasses, which is also the most complicated place here.